tools/rectool/main.c

```diff
@@ -468,6 +473,29 @@ int main(int argc, char *argv[]) {
             offset++;
         }
         printf("Deleted %d moves, final move count %d\n", offset, rec.move_count);
+    } else if(truncate->count > 0) {
+        unsigned truncate_tick = truncate->ival[0] < 0 ? 0 : truncate->ival[0];
```

Comment:
You could also just fail if a negative number was supplied instead of silently turning it into 0, which will truncate all the entries?

Reply:
yeah. that was part of the "truncate inserts a stop action" behavior i just scrapped